rust/lsp-lib/src/lsp_plugin.rs

```diff
@@ -14,23 +14,34 @@
 
 //! Implementation of Language Server Plugin
 
-use result_queue::ResultQueue;
 use conversion_utils::*;
 use language_server_client::LanguageServerClient;
 use lsp_types::*;
+use result_queue::ResultQueue;
 use serde_json;
 use std::collections::HashMap;
 use std::path::Path;
 use std::sync::Arc;
 use std::sync::Mutex;
-use types::{Config, LspResponse, LanguageResponseError};
+use types::{Config, Definition, LanguageResponseError, LspResponse};
 use url::Url;
 use utils::*;
 use xi_core::ConfigTable;
 use xi_core::ViewId;
-use xi_plugin_lib::{ChunkCache, CoreProxy, Plugin, View};
+use xi_plugin_lib::{ChunkCache, CoreProxy, Plugin, PluginContext, View};
 use xi_rope::rope::RopeDelta;
 
+/// Convenience for unwrapping a view, when handling RPC notifications.
+macro_rules! bail {
+    ($opt:expr, $view:expr) => ( match $opt {
+        Some(t) => t,
+        None => {
+            eprintln!("Path undefined for View {:?}", $view);
+            return
+        }
+    })
+}
```

Comment:
Bail does not seem like an intuitive name for this but

> nit picking

so maybe it's just me

Reply:
My main issue is that this also does a match and only conditionally bails out.